### PR TITLE
chore(demos): add common entry for lv_demos

### PR DESCRIPTION
**Pull request by bjsylvia**
### Description of the feature or fix

Provide a common entry to show all LVGL demos. 

### Checkpoints
- [ ] Run `code-format.py` from the scripts folder. [astyle](http://astyle.sourceforge.net/install.html) needs to be installed.
- [ ] Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed
- [ ] Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant. 
- [ ] Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- [ ] If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/release/v8.3/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/release/v8.3/Kconfig).
 
Be sure the following conventions are followed:
- [ ] Follow the [Styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [ ] Prefer `enum`s instead of macros. If inevitable to use `define`s export them with `LV_EXPORT_CONST_INT(defined_value)` right after the `define`.
- [ ] In function arguments prefer `type name[]` declaration for array parameters instead of `type * name`
- [ ] Use typed pointers instead of `void *` pointers
- [ ] Do not `malloc` into a static or global variables. Instead declare the variable in `lv_global_t` structure in [`lv_global.h`](https://github.com/lvgl/lvgl/blob/master/src/core/lv_global.h) and mark the variable with `(LV_GLOBAL_DEFAULT()->variable)` when it's used. See a detailed description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#memory-management).
- [ ] Widget constructor must follow the `lv_<widget_name>_create(lv_obj_t * parent)` pattern.
- [ ] Widget members function must start with `lv_<modul_name>` and should receive `lv_obj_t *` as first argument which is a pointer to widget object itself.  
- [ ] `struct`s should be used via an API and not modified directly via their elements.
- [ ] `struct` APIs should follow the widgets' conventions. That is to receive a pointer to the `struct` as the first argument, and the prefix of the `struct` name should be used as the prefix of the function name too (e.g.  `lv_disp_set_default(lv_disp_t * disp)`)
- [ ] Functions and `struct`s which are not part of the public API must begin with underscore in order to mark them as "private".
- [ ] Arguments must be named in H files too.
- [ ] To register and use callbacks one of the following needs to be followed (see a detailed description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#callbacks)):
  - For both the registration function and the callback pass a pointer to a `struct` as the first argument. The `struct` must contain `void * user_data` field.
  - The last argument of the registration function must be `void * user_data` and the same `user_data` needs to be passed as the last argument of the callback.
  - Callback types not following these conventions should end with `xcb_t`.

### CODE REVIEW

**Comment by kisvegabor** [comment]
What is the exact purpose of this feature? Why is it useful?

**Comment by bjsylvia** [comment]
LVGL provides multiple application reference examples in the `demos` directory. If there is a unified runtime entry that allows for dynamic startup of each demo, it would greatly assist in the porting and validation of LVGL on different platforms.

For example, compiling the `lv_demos.c` into an application called `lv_demos`. Then we can launch the widgets demo by `lv_demos widgets` and launch the benchmark demo by `lv_demos benchmark 1`.

**Comment by kisvegabor** [comment]
I see and it makes sense. Could you add a README.md file next to  lv_demos.c/h to explain it's purpose and usage?

**Comment by bjsylvia** [comment]
> I see and it makes sense. Could you add a README.md file next to lv_demos.c/h to explain its purpose and usage?

I've updated demos/README.md, please review it.

File: demos/lv_demos.c [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+typedef struct  {
+    const char * name;
+    union {
+        demo_method_cb entry_cb;
+#if LV_USE_DEMO_BENCHMARK
+        demo_method_benchmark_cb entry_benchmark_cb;
+        demo_method_benchmark_scene_cb entry_benchmark_scene_cb;
+#endif
+    };
+    int type : 8;
```
Comment (xiaoxiang781216) [inline demos/lv_demos.c]: remove type, arg_count is enough

File: demos/lv_demos.c [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+        usleep(1 * 1000);
+    }
+}
+#endif
+
+void lv_demos_show_usage(void)
+{
+    int i;
+    const int len = sizeof(demos_entry_info)
+                    / sizeof(demo_entry_info_t) - 1;
```
Comment (xiaoxiang781216) [inline demos/lv_demos.c]: let's remove the last empty entry

File: demos/lv_demos.c [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+        printf("lv_demos: no demo available!\n");
+        return;
+    }
+
+    printf("\nUsage: lv_demos demo [parameters]\n");
+    printf("\ndemo list:\n");
+
+    for(i = 0; i < len; i++) {
+        if(demos_entry_info[i].type == DEMO_TYPE_VOID)
+            printf("  %s \n", demos_entry_info[i].name);
```
Comment (xiaoxiang781216) [inline demos/lv_demos.c]: lv_log

File: demos/lv_demos.h [hunk 4]
```diff
@@ -68,6 +68,46 @@ extern "C" {
...
+ * Call lv_demo_xxx.
+ * @param   info the information which contains demo name and parameters needs by lv_demo_xxx.
+ * @size    size of information.
+ */
+bool lv_demos_create_demo(char * info[], int size);
+
+/**
+ * Show usage for lv_demos.
+ */
+void lv_demos_show_usage(void);
```
Comment (xiaoxiang781216) [inline demos/lv_demos.h]: lv_demos_usage

File: demos/lv_demos.h [hunk 5]
```diff
@@ -68,6 +68,46 @@ extern "C" {
  *      MACROS
  **********************/
 
+/**
+ * Call lv_demo_xxx.
+ * @param   info the information which contains demo name and parameters needs by lv_demo_xxx.
+ * @size    size of information.
+ */
+bool lv_demos_create_demo(char * info[], int size);
```
Comment (xiaoxiang781216) [inline demos/lv_demos.h]: lv_demos_create(char *argc, int argc)

File: demos/lv_demos.c (hunk 2 above)
Comment (bjsylvia) [reply demos/lv_demos.c]: should add it to avoid compiling warning

File: demos/lv_demos.h [hunk 6]
```diff
@@ -63,6 +63,18 @@ extern "C" {
...
+ * @param   info the information which contains demo name and parameters
+ *               needs by lv_demo_xxx.
+ * @size    size of information.
+ */
+bool lv_demos_create(char * info[], int size);
+
+/**
+ * Show usage for lv_demos.
+ */
+void lv_demos_usage(void);
```
Comment (kisvegabor) [inline demos/lv_demos.h]: What about this?
```suggestion
void lv_demos_show_help(void);
```

**Review by kisvegabor — COMMENTED** [review]
Thank you!

**Comment by kisvegabor** [comment]
Looks good to me! Let me know when it's ready to merge!

**Comment by bjsylvia** [comment]
> Looks good to me! Let me know when it's ready to merge!

It's ready. Thanks.

**Comment by kisvegabor** [comment]
Okkay, merging, thanks!

**Pull request merged**